File: wallet/src/main/java/chatox/wallet/config/SecurityConfig.java
package chatox.wallet.config;

import chatox.platform.security.web.AuthenticationHolder;
import chatox.platform.security.web.DefaultAuthenticationHolder;
import chatox.platform.security.web.config.AbstractSecurityConfig;
import chatox.wallet.model.User;
import chatox.wallet.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;

@Configuration
@EnableWebSecurity
@EnableMethodSecurity
public class SecurityConfig extends AbstractSecurityConfig<User> {
    @Autowired
    private UserService userService;

    @Override
    @Bean
    public AuthenticationHolder<User> authenticationHolder() {
        return new DefaultAuthenticationHolder<>(jwtPayload -> userService.findUserById(jwtPayload.getId()));
    }
}

File: chat/src/main/kotlin/chatox/chat/config/SecurityConfig.kt
package chatox.chat.config

import chatox.chat.model.User
import chatox.platform.cache.ReactiveRepositoryCacheWrapper
import chatox.platform.security.reactive.DefaultReactiveAuthenticationHolder
import chatox.platform.security.reactive.ReactiveAuthenticationHolder
import chatox.platform.security.reactive.config.AbstractReactiveSecurityConfig
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.security.config.annotation.method.configuration.EnableReactiveMethodSecurity
import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity

@Configuration
@EnableWebFluxSecurity
@EnableReactiveMethodSecurity
class SecurityConfig : AbstractReactiveSecurityConfig<User>() {
    @Autowired
    private lateinit var userCache: ReactiveRepositoryCacheWrapper<User, String>

    @Bean
    override fun reactiveAuthenticationHolder(): ReactiveAuthenticationHolder<User> = DefaultReactiveAuthenticationHolder {
        jwt -> userCache.findById(jwt.id)
    }
}
